app/src/main/java/de/danoeh/antennapod/fragment/DownloadLogFragment.java

```diff
@@ -178,6 +172,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 case R.id.clear_history_item:
                     DBWriter.clearDownloadLog();
                     return true;
+                case R.id.refresh_item:
+                    adapter.retryAll();
```

Comment:
This is executed in the main thread. If you have a lot of items, this blocks the whole app for a few seconds. Please do it in a background thread using rxjava. Untested example below.

```
Completable.fromAction(() -> adapter.retryAll())
        .subscribeOn(Schedulers.io())
        .subscribe(() -> /* TODO: Show success toast */, error -> error.printStackTrace());
```